connector_magento/models/sale_order/importer.py

```diff
@@ -0,0 +1,884 @@
+        discount = 0
+        if discount_value > 0 and row_total > 0:
+            discount = 100 * discount_value / row_total
+        return {"discount": discount}
+
+    @mapping
+    def product_id(self, record):
+        binder = self.binder_for("magento.product.product")
+        if self.collection.version == "1.7":
+            key = "product_id"
+        else:
+            key = "sku"
+        product = binder.to_internal(record[key], unwrap=True)
+        assert product, (
+            "product_id %s should have been imported in "
+            "SaleOrderImporter._import_dependencies" % record[key]
+        )
+        return {"product_id": product.id}
+
+    @mapping
+    def product_options(self, record):
+        if self.collection.version == "2.0":
+            # Product options look like this in Magento 2.0, so we'd have to
+            # fetch the labels separately -> TODO
+            # 'product_option': {
+            #     'extension_attributes': {
+            #         'configurable_item_options': [
+            #             {'option_id': '152',
+            #              'option_value': 5593},
+            #             {'option_id': '93', 'option_value': 5477}
+            #         ]
+            #     }
+            # }
+            if (
+                record.get("product_option", {})
+                .get("extension_attributes", {})
+                .get("configurable_item_options")
+            ):
+                _logger.debug(
+                    "Magento order#%s contains a product with configurable "
+                    "options but their import is not supported yet for "
+                    "Magento2"
+                )
+            return
+        result = {}
+        ifield = record["product_options"]
+        if ifield:
+            import re
+
+            options_label = []
+            clean = re.sub(r"\w:\w:|\w:\w+;", "", ifield)
+            for each in clean.split("{"):
+                if each.startswith('"label"'):
+                    split_info = each.split(";")
+                    options_label.append(
+                        "{}: {} [{}]".format(
+                            split_info[1], split_info[3], record["sku"]
+                        )
+                    )
+            notes = "".join(options_label).replace('""', "\n").replace('"', "")
+            result = {"notes": notes}
+        return result
+
+    @mapping
+    def tax_id(self, record):
+        tax_percent = float(record["tax_percent"] or 0)
+        if tax_percent > 0:
+            tax_record = self.env["account.tax"].search(
+                [
+                    ("type_tax_use", "=", "sale"),
+                    ("amount_type", "=", "percent"),
+                    ("amount", "=", tax_percent),
+                ]
+            )
+            if not tax_record:
+                raise FailedJobError("Missing tax for amount: %s" % tax_percent)
+            return {"tax_id": [(4, tax_record.id)]}
```

Comment:
We faced an issue with tax mapping as previous version a tax was set from the product settings and then a fiscal position was applied that handles correctly final taxes.

This code tries to take a tax with the same percentage even if the tax on a product is different. Extra, in case there are more than one tax (for different countries) with the same percentage it will fail as code expect to provide 1 tax, not all of them.

My suggestion is to remove this mapping and leave Odoo default behaviour when a tax is set on the SO line.